Hardening: Use ThreadSafeWeakPtrs in VideoFullscreenInterface{Mac,AVKit}

jyavenard opened the pull request:
#### 9a1b8a792aee871c2564489d255b19cd9d218e22
<pre>
Hardening: Use ThreadSafeWeakPtrs in VideoFullscreenInterface{Mac,AVKit}
<a href="https://bugs.webkit.org/show_bug.cgi?id=258025">https://bugs.webkit.org/show_bug.cgi?id=258025</a>
rdar://110311635

Reviewed by Youenn Fablet.

The VideoFullscreenInterface{Mac,AVKit} implementations store their fullscreen model
and fullscreen change observer members as bare pointers. The use of WeakPtr
wasn&apos;t possible at the time (see webkit.org/b/199698) as the objects are
allocated on the UIThread and use on the WebThread.
So we use the new ThreadSafeWeakPtr instead.
VideoFullscreenInterface{Mac,AVKit} alwas referred to its VideoFullscreenChangeObserver
and VideoFullscreenModel as a pair and always using a single class implementing
both protocols (respectively VideoFullscreenControllerContext and VideoFullscreenModelContext).
We merge the two into a VideoFullscreenModel which now implements
`ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr`.
This solves two problems:
1- there&apos;s complexity in implementing a class that inherits from two classes
implementing the same methods (ref()/deref())
2- It simplifies the logic used.

Fly-by: moves default implementation of EventListener::operator==.

No new tests since no changes in behavior.

* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
* Source/WebCore/PlatformMac.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/EventListener.h:
(WebCore::EventListener::operator== const):
* Source/WebCore/html/HTMLAttachmentElement.cpp:
* Source/WebCore/html/HTMLMediaElement.h:
(WebCore::HTMLMediaElement::waitForPreparedForInlineThen): Deleted.
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
* Source/WebCore/inspector/agents/InspectorIndexedDBAgent.cpp:
* Source/WebCore/platform/cocoa/PlaybackSessionModelMediaElement.h:
* Source/WebCore/platform/cocoa/VideoFullscreenChangeObserver.h: Removed.
* Source/WebCore/platform/cocoa/VideoFullscreenModel.h:
(WebCore::VideoFullscreenModel::requestUpdateInlineRect):
(WebCore::VideoFullscreenModel::requestVideoContentLayer):
(WebCore::VideoFullscreenModel::returnVideoContentLayer):
(WebCore::VideoFullscreenModel::returnVideoView):
(WebCore::VideoFullscreenModel::didSetupFullscreen):
(WebCore::VideoFullscreenModel::didEnterFullscreen):
(WebCore::VideoFullscreenModel::failedToEnterFullscreen):
(WebCore::VideoFullscreenModel::willExitFullscreen):
(WebCore::VideoFullscreenModel::didExitFullscreen):
(WebCore::VideoFullscreenModel::didCleanupFullscreen):
(WebCore::VideoFullscreenModel::fullscreenMayReturnToInline):
(WebCore::VideoFullscreenModelClient::didExitPictureInPicture):
(WebCore::VideoFullscreenModel::~VideoFullscreenModel): Deleted.
(WebCore::VideoFullscreenModelClient::modelDestroyed): Deleted.
* Source/WebCore/platform/cocoa/VideoFullscreenModelVideoElement.h:
(WebCore::VideoFullscreenModelVideoElement::create): Deleted.
(WebCore::VideoFullscreenModelVideoElement::videoElement const): Deleted.
(WebCore::VideoFullscreenModelVideoElement::setVideoFullscreenLayer): Deleted.
(WebCore::VideoFullscreenModelVideoElement::waitForPreparedForInlineThen): Deleted.
(WebCore::VideoFullscreenModelVideoElement::logClassName const): Deleted.
* Source/WebCore/platform/cocoa/VideoFullscreenModelVideoElement.mm:
(WebCore::VideoFullscreenModelVideoElement::VideoListener::VideoListener):
(WebCore::VideoFullscreenModelVideoElement::VideoListener::handleEvent):
(WebCore::VideoFullscreenModelVideoElement::VideoFullscreenModelVideoElement):
(WebCore::VideoFullscreenModelVideoElement::~VideoFullscreenModelVideoElement):
(WebCore::VideoFullscreenModelVideoElement::cleanVideoListeners):
(WebCore::VideoFullscreenModelVideoElement::setVideoElement):
(WebCore::VideoFullscreenModelVideoElement::handleEvent): Deleted.
* Source/WebCore/platform/cocoa/WebAVPlayerLayer.mm:
(-[WebAVPlayerLayer dealloc]):
(-[WebAVPlayerLayer fullscreenModel]):
(-[WebAVPlayerLayer setFullscreenModel:]):
(-[WebAVPlayerLayer resolveBounds]):
(-[WebAVPlayerLayer setVideoGravity:]):
(-[WebAVPlayerLayer loggerPtr]):
* Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.h:
* Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm:
(VideoFullscreenInterfaceAVKit::~VideoFullscreenInterfaceAVKit):
(VideoFullscreenInterfaceAVKit::setVideoFullscreenModel):
(VideoFullscreenInterfaceAVKit::presentingViewController):
(VideoFullscreenInterfaceAVKit::cleanupFullscreen):
(VideoFullscreenInterfaceAVKit::invalidate):
(VideoFullscreenInterfaceAVKit::requestHideAndExitFullscreen):
(VideoFullscreenInterfaceAVKit::preparedToExitFullscreen):
(VideoFullscreenInterfaceAVKit::prepareForPictureInPictureStop):
(VideoFullscreenInterfaceAVKit::willStartPictureInPicture):
(VideoFullscreenInterfaceAVKit::failedToStartPictureInPicture): fly-by fix: if we failed to enter PiP
 then there would be no need to return to standby fullscreen. So set flag accordingly.
(VideoFullscreenInterfaceAVKit::willStopPictureInPicture):
(VideoFullscreenInterfaceAVKit::didStopPictureInPicture):
(VideoFullscreenInterfaceAVKit::shouldExitFullscreenWithReason):
(VideoFullscreenInterfaceAVKit::doSetup):
(VideoFullscreenInterfaceAVKit::finalizeSetup):
(VideoFullscreenInterfaceAVKit::doEnterFullscreen):
(VideoFullscreenInterfaceAVKit::doExitFullscreen):
(VideoFullscreenInterfaceAVKit::returnToStandby):
(VideoFullscreenInterfaceAVKit::setMode):
(VideoFullscreenInterfaceAVKit::clearMode):
(VideoFullscreenInterfaceAVKit::setVideoFullscreenChangeObserver): Deleted.
(VideoFullscreenInterfaceAVKit::modelDestroyed): Deleted.
* Source/WebCore/platform/ios/WebVideoFullscreenControllerAVKit.mm:
(VideoFullscreenControllerContext::~VideoFullscreenControllerContext):
(VideoFullscreenControllerContext::didCleanupFullscreen):
(VideoFullscreenControllerContext::setUpFullscreen):
* Source/WebCore/platform/mac/VideoFullscreenInterfaceMac.h:
(WebCore::VideoFullscreenInterfaceMac::videoFullscreenModel const):
(WebCore::VideoFullscreenInterfaceMac::videoFullscreenChangeObserver const): Deleted.
* Source/WebCore/platform/mac/VideoFullscreenInterfaceMac.mm:
(-[WebVideoFullscreenInterfaceMacObjC setUpPIPForVideoView:withFrame:inWindow:]):
(-[WebVideoFullscreenInterfaceMacObjC boundsDidChangeForVideoViewContainer:]):
(-[WebVideoFullscreenInterfaceMacObjC pipShouldClose:]):
(-[WebVideoFullscreenInterfaceMacObjC pipDidClose:]):
(WebCore::VideoFullscreenInterfaceMac::~VideoFullscreenInterfaceMac):
(WebCore::VideoFullscreenInterfaceMac::setVideoFullscreenModel):
(WebCore::VideoFullscreenInterfaceMac::setMode):
(WebCore::VideoFullscreenInterfaceMac::clearMode):
(WebCore::VideoFullscreenInterfaceMac::setupFullscreen):
(WebCore::VideoFullscreenInterfaceMac::enterFullscreen):
(WebCore::VideoFullscreenInterfaceMac::cleanupFullscreen):
(WebCore::VideoFullscreenInterfaceMac::invalidate):
(WebCore::VideoFullscreenInterfaceMac::requestHideAndExitPiP):
(WebCore::VideoFullscreenInterfaceMac::setVideoFullscreenChangeObserver): Deleted.
* Source/WebKit/UIProcess/Cocoa/VideoFullscreenManagerProxy.h:
* Source/WebKit/UIProcess/Cocoa/VideoFullscreenManagerProxy.mm:
(WebKit::VideoFullscreenManagerProxy::createModelAndInterface):
(WebKit::VideoFullscreenModelContext::~VideoFullscreenModelContext): Deleted.
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(-[WKFullScreenWindowController beganEnterFullScreenWithInitialFrame:finalFrame:]):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginAnnotation.h:
(): Deleted.
* Source/WebKitLegacy/mac/DOM/ObjCEventListener.h: use override keywords.

Canonical link: <a href="https://commits.webkit.org/265195@main">https://commits.webkit.org/265195@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cc081bd767a311cbfae8c92af44bbce8e32633e1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10210 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/10448 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/10710 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/11856 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/9846 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/10222 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/12442 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10400 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/12782 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10368 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/11113 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8608 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12239 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/8418 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9248 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/16537 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/9517 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9400 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12644 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9857 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/8000 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9016 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2442 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13264 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9684 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/platform/cocoa/VideoFullscreenModelVideoElement.h:
```diff
@@ -47,23 +47,19 @@ class HTMLVideoElement;
-    WEBCORE_EXPORT void handleEvent(WebCore::ScriptExecutionContext&, WebCore::Event&) override;
-    void updateForEventName(const AtomString&);
-    bool operator==(const EventListener& rhs) const override { return static_cast<const WebCore::EventListener*>(this) == &rhs; }
+    WEBCORE_EXPORT void willExitFullscreen() override;
```
Given the class is final, we probably want to use final in this method (and others as well).

jyavenard replied:
done

youennf commented on Source/WebCore/platform/cocoa/VideoFullscreenModelVideoElement.h:
```diff
@@ -90,6 +86,20 @@ class VideoFullscreenModelVideoElement : public VideoFullscreenModel, public Eve
+        {
+            return adoptRef(*new VideoListener(parent));
+        }
+        void handleEvent(WebCore::ScriptExecutionContext&, WebCore::Event&) override;
```
s/final/override

youennf commented on Source/WebCore/platform/cocoa/VideoFullscreenModelVideoElement.h:
```diff
@@ -90,6 +86,20 @@ class VideoFullscreenModelVideoElement : public VideoFullscreenModel, public Eve
 private:
+    class VideoListener final : public EventListener {
+    public:
+        static Ref<VideoListener> create(VideoFullscreenModelVideoElement* parent)
```
Would be better and cheaper to pass a ref (assuming parent here is probably never null).

youennf replied:
Looking at VideoListener::create in this patch, it seems it is given `this` (maybe I missed some code paths though).
If so, it seems we could pass `*this` so that the creation of the weak pointer is made cheaper as this would remove a nullptr check.

jyavenard replied:
yes

youennf commented on Source/WebCore/platform/cocoa/VideoFullscreenModelVideoElement.h:
```diff
@@ -90,6 +86,20 @@ class VideoFullscreenModelVideoElement : public VideoFullscreenModel, public Eve
+            return adoptRef(*new VideoListener(parent));
+        }
+        void handleEvent(WebCore::ScriptExecutionContext&, WebCore::Event&) override;
+        bool operator==(const EventListener& rhs) const override { return static_cast<const WebCore::EventListener*>(this) == &rhs; }
```
Is `return this == &rhs;` not working?
Maybe this implementation should be the default one.

jyavenard replied:
I copied/paste the code, not looking much. the method used to inherit from two classes. I wonder why this even needs an override tbh

youennf commented on Source/WebCore/platform/cocoa/VideoFullscreenModelVideoElement.h:
```diff
@@ -90,6 +86,20 @@ class VideoFullscreenModelVideoElement : public VideoFullscreenModel, public Eve
+        void handleEvent(WebCore::ScriptExecutionContext&, WebCore::Event&) override;
+        bool operator==(const EventListener& rhs) const override { return static_cast<const WebCore::EventListener*>(this) == &rhs; }
+    private:
+        VideoListener(VideoFullscreenModelVideoElement*);
```
explicit and VideoFullscreenModelVideoElement&

youennf commented on Source/WebCore/platform/cocoa/VideoFullscreenModelVideoElement.mm:
```diff
@@ -49,13 +49,34 @@
+}
+
+VideoFullscreenModelVideoElement::VideoFullscreenModelVideoElement()
+    : m_videoListener(VideoListener::create(this))
```
*this then

youennf commented on Source/WebCore/platform/cocoa/VideoFullscreenModelVideoElement.mm:
```diff
@@ -49,13 +49,34 @@
+
+void VideoFullscreenModelVideoElement::cleanVideoListeners()
+{
+    if (!m_videoElement)
```
Can we add:
```
if (! m_isListening)
   return;
m_isListening = false;
```

youennf commented on Source/WebCore/platform/cocoa/WebAVPlayerLayer.mm:
```diff
@@ -106,32 +105,33 @@ - (void)dealloc
-    if (_fullscreenModel)
-        _fullscreenModel->removeClient(*_fullscreenModelClient);
+    if (auto model = _fullscreenModel.get())
+        model->removeClient(*_fullscreenModelClient);
```
_fullscreenModelClient could become a UniqueRef.

jyavenard replied:
it's obj-C, there's no initialiser that would be able to set it up

youennf commented on Source/WebCore/platform/cocoa/WebAVPlayerLayer.mm:
```diff
@@ -106,32 +105,33 @@ - (void)dealloc
 - (VideoFullscreenModel*)fullscreenModel
 {
-    return _fullscreenModel.get();
+    return _fullscreenModel.get().get();
```
Could we refactor code to remove this getter?
Looking quickly at the code base, it seems used in only one place.

youennf commented on Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm:
```diff
@@ -769,14 +767,16 @@ - (WTFLogChannel*)logChannel
-        m_videoFullscreenModel->removeClient(*this);
+    auto oldModel = videoFullscreenModel();
+
+    if (oldModel)
```
```suggestion
    if (auto oldModel = videoFullscreenModel())
```

youennf commented on Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm:
```diff
@@ -1028,9 +1019,9 @@ - (WTFLogChannel*)logChannel
-    if (m_videoFullscreenModel)
-        m_videoFullscreenModel->requestFullscreenMode(HTMLMediaElementEnums::VideoFullscreenModeNone, true);
+    auto model = videoFullscreenModel();
+    if (model)
```
```suggestion
    if (auto model = videoFullscreenModel())
```

youennf commented on Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm:
```diff
@@ -1096,15 +1088,14 @@ - (WTFLogChannel*)logChannel
-        m_fullscreenChangeObserver->failedToEnterFullscreen();
-        m_changingStandbyOnly = false;
+    auto model = videoFullscreenModel();
+    if (model) {
```
```suggestion
    if (auto model = videoFullscreenModel()) {
```

youennf commented on Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm:
```diff
@@ -1096,15 +1088,14 @@ - (WTFLogChannel*)logChannel
+        model->fullscreenModeChanged(HTMLMediaElementEnums::VideoFullscreenModeNone);
+        model->failedToEnterFullscreen();
     }
+    m_changingStandbyOnly = false;
```
Why is this change needed now?

jyavenard replied:
it was an error that it was only set if the observer was null. We failed to enter PiP, so there's no need to return to standby fullscreen, we are already in that mode.

jyavenard replied:
I will add it in the commit log.

youennf commented on Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm:
```diff
@@ -1227,19 +1219,17 @@ - (WTFLogChannel*)logChannel
+    return false;
+#else
 
     BOOL finished = reason == ExitFullScreenReason::DoneButtonTapped || reason == ExitFullScreenReason::PinchGestureHandled;
```
Why these empty lines here and below?

youennf commented on Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm:
```diff
@@ -1478,18 +1470,19 @@ - (WTFLogChannel*)logChannel
 
-    if (m_hasVideoContentLayer && m_fullscreenChangeObserver) {
+    if (m_hasVideoContentLayer) {
         m_exitFullscreenNeedsReturnContentLayer = true;
```
We were not setting it back to true previously.

jyavenard replied:
yeah, I think it could have introduced an error. thanks

jyavenard committed:
Hardening: Use ThreadSafeWeakPtrs in VideoFullscreenInterface{Mac,AVKit}
https://bugs.webkit.org/show_bug.cgi?id=258025
rdar://110311635

Reviewed by Youenn Fablet.

The VideoFullscreenInterface{Mac,AVKit} implementations store their fullscreen model
and fullscreen change observer members as bare pointers. The use of WeakPtr
wasn't possible at the time (see webkit.org/b/199698) as the objects are
allocated on the UIThread and use on the WebThread.
So we use the new ThreadSafeWeakPtr instead.
VideoFullscreenInterface{Mac,AVKit} alwas referred to its VideoFullscreenChangeObserver
and VideoFullscreenModel as a pair and always using a single class implementing
both protocols (respectively VideoFullscreenControllerContext and VideoFullscreenModelContext).
We merge the two into a VideoFullscreenModel which now implements
`ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr`.
This solves two problems:
1- there's complexity in implementing a class that inherits from two classes
implementing the same methods (ref()/deref())
2- It simplifies the logic used.

Fly-by: moves default implementation of EventListener::operator==.

No new tests since no changes in behavior.

* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
* Source/WebCore/PlatformMac.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/EventListener.h:
(WebCore::EventListener::operator== const):
* Source/WebCore/html/HTMLAttachmentElement.cpp:
* Source/WebCore/html/HTMLMediaElement.h:
(WebCore::HTMLMediaElement::waitForPreparedForInlineThen): Deleted.
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
* Source/WebCore/inspector/agents/InspectorIndexedDBAgent.cpp:
* Source/WebCore/platform/cocoa/PlaybackSessionModelMediaElement.h:
* Source/WebCore/platform/cocoa/VideoFullscreenChangeObserver.h: Removed.
* Source/WebCore/platform/cocoa/VideoFullscreenModel.h:
(WebCore::VideoFullscreenModel::requestUpdateInlineRect):
(WebCore::VideoFullscreenModel::requestVideoContentLayer):
(WebCore::VideoFullscreenModel::returnVideoContentLayer):
(WebCore::VideoFullscreenModel::returnVideoView):
(WebCore::VideoFullscreenModel::didSetupFullscreen):
(WebCore::VideoFullscreenModel::didEnterFullscreen):
(WebCore::VideoFullscreenModel::failedToEnterFullscreen):
(WebCore::VideoFullscreenModel::willExitFullscreen):
(WebCore::VideoFullscreenModel::didExitFullscreen):
(WebCore::VideoFullscreenModel::didCleanupFullscreen):
(WebCore::VideoFullscreenModel::fullscreenMayReturnToInline):
(WebCore::VideoFullscreenModelClient::didExitPictureInPicture):
(WebCore::VideoFullscreenModel::~VideoFullscreenModel): Deleted.
(WebCore::VideoFullscreenModelClient::modelDestroyed): Deleted.
* Source/WebCore/platform/cocoa/VideoFullscreenModelVideoElement.h:
(WebCore::VideoFullscreenModelVideoElement::create): Deleted.
(WebCore::VideoFullscreenModelVideoElement::videoElement const): Deleted.
(WebCore::VideoFullscreenModelVideoElement::setVideoFullscreenLayer): Deleted.
(WebCore::VideoFullscreenModelVideoElement::waitForPreparedForInlineThen): Deleted.
(WebCore::VideoFullscreenModelVideoElement::logClassName const): Deleted.
* Source/WebCore/platform/cocoa/VideoFullscreenModelVideoElement.mm:
(WebCore::VideoFullscreenModelVideoElement::VideoListener::VideoListener):
(WebCore::VideoFullscreenModelVideoElement::VideoListener::handleEvent):
(WebCore::VideoFullscreenModelVideoElement::VideoFullscreenModelVideoElement):
(WebCore::VideoFullscreenModelVideoElement::~VideoFullscreenModelVideoElement):
(WebCore::VideoFullscreenModelVideoElement::cleanVideoListeners):
(WebCore::VideoFullscreenModelVideoElement::setVideoElement):
(WebCore::VideoFullscreenModelVideoElement::handleEvent): Deleted.
* Source/WebCore/platform/cocoa/WebAVPlayerLayer.mm:
(-[WebAVPlayerLayer dealloc]):
(-[WebAVPlayerLayer fullscreenModel]):
(-[WebAVPlayerLayer setFullscreenModel:]):
(-[WebAVPlayerLayer resolveBounds]):
(-[WebAVPlayerLayer setVideoGravity:]):
(-[WebAVPlayerLayer loggerPtr]):
* Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.h:
* Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm:
(VideoFullscreenInterfaceAVKit::~VideoFullscreenInterfaceAVKit):
(VideoFullscreenInterfaceAVKit::setVideoFullscreenModel):
(VideoFullscreenInterfaceAVKit::presentingViewController):
(VideoFullscreenInterfaceAVKit::cleanupFullscreen):
(VideoFullscreenInterfaceAVKit::invalidate):
(VideoFullscreenInterfaceAVKit::requestHideAndExitFullscreen):
(VideoFullscreenInterfaceAVKit::preparedToExitFullscreen):
(VideoFullscreenInterfaceAVKit::prepareForPictureInPictureStop):
(VideoFullscreenInterfaceAVKit::willStartPictureInPicture):
(VideoFullscreenInterfaceAVKit::failedToStartPictureInPicture): fly-by fix: if we failed to enter PiP
 then there would be no need to return to standby fullscreen. So set flag accordingly.
(VideoFullscreenInterfaceAVKit::willStopPictureInPicture):
(VideoFullscreenInterfaceAVKit::didStopPictureInPicture):
(VideoFullscreenInterfaceAVKit::shouldExitFullscreenWithReason):
(VideoFullscreenInterfaceAVKit::doSetup):
(VideoFullscreenInterfaceAVKit::finalizeSetup):
(VideoFullscreenInterfaceAVKit::doEnterFullscreen):
(VideoFullscreenInterfaceAVKit::doExitFullscreen):
(VideoFullscreenInterfaceAVKit::returnToStandby):
(VideoFullscreenInterfaceAVKit::setMode):
(VideoFullscreenInterfaceAVKit::clearMode):
(VideoFullscreenInterfaceAVKit::setVideoFullscreenChangeObserver): Deleted.
(VideoFullscreenInterfaceAVKit::modelDestroyed): Deleted.
* Source/WebCore/platform/ios/WebVideoFullscreenControllerAVKit.mm:
(VideoFullscreenControllerContext::~VideoFullscreenControllerContext):
(VideoFullscreenControllerContext::didCleanupFullscreen):
(VideoFullscreenControllerContext::setUpFullscreen):
* Source/WebCore/platform/mac/VideoFullscreenInterfaceMac.h:
(WebCore::VideoFullscreenInterfaceMac::videoFullscreenModel const):
(WebCore::VideoFullscreenInterfaceMac::videoFullscreenChangeObserver const): Deleted.
* Source/WebCore/platform/mac/VideoFullscreenInterfaceMac.mm:
(-[WebVideoFullscreenInterfaceMacObjC setUpPIPForVideoView:withFrame:inWindow:]):
(-[WebVideoFullscreenInterfaceMacObjC boundsDidChangeForVideoViewContainer:]):
(-[WebVideoFullscreenInterfaceMacObjC pipShouldClose:]):
(-[WebVideoFullscreenInterfaceMacObjC pipDidClose:]):
(WebCore::VideoFullscreenInterfaceMac::~VideoFullscreenInterfaceMac):
(WebCore::VideoFullscreenInterfaceMac::setVideoFullscreenModel):
(WebCore::VideoFullscreenInterfaceMac::setMode):
(WebCore::VideoFullscreenInterfaceMac::clearMode):
(WebCore::VideoFullscreenInterfaceMac::setupFullscreen):
(WebCore::VideoFullscreenInterfaceMac::enterFullscreen):
(WebCore::VideoFullscreenInterfaceMac::cleanupFullscreen):
(WebCore::VideoFullscreenInterfaceMac::invalidate):
(WebCore::VideoFullscreenInterfaceMac::requestHideAndExitPiP):
(WebCore::VideoFullscreenInterfaceMac::setVideoFullscreenChangeObserver): Deleted.
* Source/WebKit/UIProcess/Cocoa/VideoFullscreenManagerProxy.h:
* Source/WebKit/UIProcess/Cocoa/VideoFullscreenManagerProxy.mm:
(WebKit::VideoFullscreenManagerProxy::createModelAndInterface):
(WebKit::VideoFullscreenModelContext::~VideoFullscreenModelContext): Deleted.
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(-[WKFullScreenWindowController beganEnterFullScreenWithInitialFrame:finalFrame:]):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginAnnotation.h:
(): Deleted.
* Source/WebKitLegacy/mac/DOM/ObjCEventListener.h: use override keywords.

Canonical link: https://commits.webkit.org/265195@main
